### PR TITLE
FLU: Convert street furniture

### PR DESCRIPTION
**Pull request by lukicdarkoo**
- [x] (138) objects/street_furniture/protos/Bench.proto
- [x] (139) objects/street_furniture/protos/UrbanFence.proto
- [x] (140) objects/street_furniture/protos/BusStopBench.proto
- [x] (141) objects/street_furniture/protos/WorkTrashContainer.proto
- [x] (142) objects/street_furniture/protos/StoneBench.proto
- [x] (143) objects/street_furniture/protos/MetallicTrash.proto
- [x] (144) objects/street_furniture/protos/Mailbox.proto
- [x] (145) objects/street_furniture/protos/BusStop.proto
- [x] (146) objects/street_furniture/protos/StoneFountain.proto
- [x] (147) objects/street_furniture/protos/OldBench.proto
- [x] (148) objects/street_furniture/protos/FireHydrant.proto
- [x] (149) objects/street_furniture/protos/SimpleBench.proto
- [x] (150) objects/street_furniture/protos/TrashContainer.proto
- [x] (151) objects/street_furniture/protos/GuardShelter.proto
- [x] (152) objects/street_furniture/protos/DrinkingFountain.proto
- [x] (153) objects/street_furniture/protos/PublicToilet.proto (depends on `AdvertisingBoard`, #3639)
- [x] (154) objects/street_furniture/protos/IceFreezerContainer.proto
- [x] (155) objects/street_furniture/protos/TrashBin.proto
- [x] (156) objects/street_furniture/protos/EmergencyPhone.proto
- [x] (157) objects/street_furniture/protos/GlassRecyclingContainer.proto
- [x] (158) objects/street_furniture/protos/UndergroundContainer.proto
- [x] (159) objects/street_furniture/protos/ClothRecyclingContainer.proto
- [x] (160) objects/street_furniture/protos/PublicBin.proto
- [x] (161) objects/street_furniture/protos/Fountain.proto (depends on `SolidPipe`)
- [x] (162) objects/street_furniture/protos/ElectricalCabinet.proto
- [x] (163) objects/street_furniture/protos/NewsStand.proto
- [x] (164) objects/street_furniture/protos/PhoneBooth.proto
- [x] (165) objects/street_furniture/protos/Fence.proto
- [x] (166) objects/street_furniture/protos/SmallKiosk.proto
- [x] (167) objects/street_furniture/protos/PublicToiletMesh.proto
- [x] (168) objects/street_furniture/protos/Atm.proto
- [x] (169) objects/street_furniture/protos/SnackStand.proto
- [x] (170) objects/street_furniture/protos/BusStopMesh.proto

### CODE REVIEW

**Review by omichel — CHANGES_REQUESTED** [review]
It looks very good.
I just noticed a couple of minor problems:
- The texture of ad in the BusStop proto is not properly oriented.
- Also the X direction of the BusStop proto seems wrong.
- The XY origin of the MailBox and TrashContainer is not centered on the object, which looks strange, but maybe it was the case before FLU conversion...

File: projects/objects/street_furniture/protos/Fence.proto [hunk 1]
```diff
@@ -237,21 +236,19 @@ PROTO Fence [
...
                   coordIndexBuffer += i * skip + j + circleSubdivision + 1 + ' ';
                 coordIndexBuffer += ' -1\n';
 
                 // Sides
                 for (let j = 0; j <= circleSubdivision - 1; ++j) {
-                  coordIndexBuffer += i * skip + j + ' ';
-                  coordIndexBuffer += i * skip + j + 1 + ' ';
-                  coordIndexBuffer += i * skip + j + circleSubdivision + 2  + ' ';
-                  coordIndexBuffer += i * skip + j + circleSubdivision + 1 + ' -1\n';
+                  const start = i * skip + j;
```
Comment (ad-daniel) [inline projects/objects/street_furniture/protos/Fence.proto:250]: The Fence PROTO is especially slow in JavaScript compared to Lua, and is also used to define things like `CrashBarrier` so although template literals are more readable, it's best to build the geometry in a buffer for performance reasons. This PROTO is one among those that added a huge overhead to world like `village_center`.

Comment (lukicdarkoo) [reply projects/objects/street_furniture/protos/Fence.proto:250]: Actually, it has a performance impact :)
https://jsbench.me/0skt9pgzus/1

**Comment by lukicdarkoo** [comment]
> The XY origin of the MailBox and TrashContainer is not centered on the object, which looks strange, but maybe it was the case before FLU conversion...

It was like that before. Should we change it?

**Comment by omichel** [comment]
> It was like that before. Should we change it?

I would be nice to change it, but that would require that we carefully update all the worlds using them to ensure the objects are properly positioned.

**Review by omichel — APPROVED** [review]
It looks all good to me.
Thank you.

**Pull request merged**